### PR TITLE
make entities activitypub actors

### PR DESCRIPTION
**Pull request by maxlath**
implements:
* webfinger for entities, using entities uris as name, ex: `wd:Q535@inventaire.io`
* actor endpoint
* inbox endpoint, to receive follow, using a shared key pair
* outbox endpoint, building activities from patches creating claims with the actor entity as claim object
* activity delivery, triggered when one of those patches is created

This is currently only implemented for new author works (that is, when `wdt:P50` claims are created), as an example to validate the process, but could be extended to include other kinds of entity relations

### CODE REVIEW

File: tests/api/activitypub/followers_activity_delivery.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        url: inboxUrl,
+        object: followedActorUrl,
+        type: 'Follow',
+      })
+      await addAuthor(workUri, authorUri)
+      await wait(500)
+      const { inbox } = await requests_.get(`${remoteHost}/inbox_inspection?username=${remoteUsername}`)
+      const createActivity = inbox[0]
+      createActivity.object.content.should.containEql(authorLabels.en)
+      createActivity.object.content.should.containEql(workLabels.en)
```
Comment (jum-s) [inline tests/api/activitypub/followers_activity_delivery.test.js:49]: this rely on a up to date i18n, maybe checking if object id contains `patchId` or just the content contains `<p>` for me is enough has its the activity posting that is tested more than what is posted

File: tests/api/activitypub/outbox.test.js [hunk 2]
```diff
@@ -5,189 +5,225 @@ require('should')
...
+      decodeURIComponent(res.next).should.equal(`${fullHostUrl}&offset=10`)
+      res.orderedItems.should.be.an.Array()
+      res.orderedItems.length.should.equal(1)
+      const createActivity = res.orderedItems[0]
+      const actorUrl = makeUrl({ params: { action: 'actor', name: authorUri } })
+      const activityEndpoint = makeUrl({ params: { action: 'activity' } })
+      createActivity.id.should.startWith(activityEndpoint)
+      createActivity.actor.should.equal(actorUrl)
+      createActivity.object.content.should.containEql(authorLabels.en)
+      createActivity.object.content.should.containEql(workLabels.en)
```
Comment (jum-s) [inline tests/api/activitypub/outbox.test.js:209]: idem

File: tests/api/activitypub/outbox.test.js [hunk 3]
```diff
@@ -5,189 +5,225 @@ require('should')
...
+      await addAuthor(workUri, authorUri)
+      await addAuthor(workUri2, authorUri)
+      const outboxUrl = `${endpoint}${authorUri}`
+      await wait(500)
+      const res1 = await publicReq('get', `${outboxUrl}&offset=0&limit=1`)
+      res1.orderedItems.length.should.equal(1)
+      res1.orderedItems[0].object.content.should.containEql(workLabels2.en)
+      const res2 = await publicReq('get', `${outboxUrl}&offset=1&limit=1`)
+      res2.orderedItems.length.should.equal(1)
+      res2.orderedItems[0].object.content.should.containEql(workLabels1.en)
```
Comment (jum-s) [inline tests/api/activitypub/outbox.test.js:226]: idem twice with line 223

File: server/controllers/activitypub/outbox.js [hunk 4]
```diff
@@ -44,13 +77,27 @@ module.exports = {
...
+}
+
+const buildPaginatedEntityOutbox = async (entity, offset, limit, outbox) => {
+  const { id: fullOutboxUrl } = outbox
+  outbox.type = 'OrderedCollectionPage'
+  outbox.partOf = fullOutboxUrl
+  outbox.next = `${fullOutboxUrl}&offset=${offset + limit}`
+  // const { stableUsername } = user
+  // const activitiesDocs = await byActorName({ name: stableUsername, offset, limit })
+  // outbox.orderedItems = await formatUserItemsActivities(activitiesDocs, user)
```
Comment (jum-s) [inline server/controllers/activitypub/outbox.js:98]: obsolete comments

**Review by jum-s — CHANGES_REQUESTED** [review]

File: server/controllers/activitypub/outbox.js (hunk 4 above)
Comment (maxlath) [reply server/controllers/activitypub/outbox.js]: removed in a5786745

File: tests/api/activitypub/followers_activity_delivery.test.js (hunk 1 above)
Comment (maxlath) [reply tests/api/activitypub/followers_activity_delivery.test.js]: addressed in 7e0456cf

File: tests/api/activitypub/outbox.test.js (hunk 2 above)
Comment (maxlath) [reply tests/api/activitypub/outbox.test.js]: addressed in 7e0456cf

File: tests/api/activitypub/outbox.test.js (hunk 3 above)
Comment (maxlath) [reply tests/api/activitypub/outbox.test.js]: addressed in 7e0456cf

**Review by jum-s — APPROVED** [review]

**Comment by maxlath** [comment]
@jum-s webfinger implementation got lost in a previous rebase, now cherry-picked as 809889f5

**Pull request merged**